lbrynet/lbrynet_daemon/LBRYDaemon.py

```diff
@@ -1685,65 +1608,54 @@ def jsonrpc_resolve_name(self, p):
 
         Args:
             'name': name to download, string
             'download_directory': optional, path to directory where file will be saved, string
             'file_name': optional, a user specified name for the downloaded file
             'stream_info': optional, specified stream info overrides name
+            'timout': optional
+            'wait_for_write': optional, defaults to True
         Returns:
             'stream_hash': hex string
             'path': path of download
         """
-
-        if 'timeout' not in p.keys():
-            timeout = self.download_timeout
-        else:
-            timeout = p['timeout']
-
-        if 'download_directory' not in p.keys():
-            download_directory = self.download_directory
-        else:
-            download_directory = p['download_directory']
-
-        if 'file_name' in p.keys():
-            file_name = p['file_name']
-        else:
-            file_name = None
-
-        if 'stream_info' in p.keys():
-            stream_info = p['stream_info']
-            if 'sources' in stream_info.keys():
-                sd_hash = stream_info['sources']['lbry_sd_hash']
-            else:
-                sd_hash = stream_info['stream_hash']
-        else:
-            stream_info = None
-
-        if 'wait_for_write' in p.keys():
-            wait_for_write = p['wait_for_write']
-        else:
-            wait_for_write = True
-
-        if 'name' in p.keys():
-            name = p['name']
-            if p['name'] not in self.waiting_on.keys():
-                d = self._download_name(name=name, timeout=timeout, download_directory=download_directory,
-                                        stream_info=stream_info, file_name=file_name, wait_for_write=wait_for_write)
-                d.addCallback(lambda l: {'stream_hash': sd_hash,
-                                         'path': os.path.join(self.download_directory, l.file_name)}
-                                         if stream_info else
-                                         {'stream_hash': l.sd_hash,
-                                         'path': os.path.join(self.download_directory, l.file_name)})
-                d.addCallback(lambda message: self._render_response(message, OK_CODE))
-            else:
-                d = server.failure
-        else:
-            d = server.failure
-
+        params = self._process_get_parameters(p)
+        if not params.name:
+            return server.failure
+        if params.name in self.waiting_on:
+            return server.failure
+        d = self._download_name(name=params.name,
+                                timeout=params.timeout,
+                                download_directory=params.download_directory,
+                                stream_info=params.stream_info,
+                                file_name=params.file_name,
+                                wait_for_write=params.wait_for_write)
+        d.addCallback(get_output_callback(params))
```

Comment:
This was commented out here: https://github.com/lbryio/lbry/commit/0bee2689d1f276f5f9c86d66ba567fe85f46b766#diff-4510ef2fe0fe3e8b0ff6c011d8cc31abR1643 and I don't know why.

It matched up with https://github.com/lbryio/lbry/blob/e67f55bd7447cbee730d895b7f35cf53cf69dba5/lbrynet/lbrynet_daemon/LBRYDaemon.py#L1736